steel/tests/view_call.rs

```diff
@@ -249,14 +267,39 @@ fn call_eoa() {
     let mut env = EthViewCallEnv::from_provider(provider, VIEW_CALL_TEST_BLOCK)
         .unwrap()
         .with_chain_spec(&ETH_SEPOLIA_CHAIN_SPEC);
-    env.preflight(ViewCall::new(
-        ViewCallTest::testBlockhashCall {},
-        Address::ZERO,
-    ))
-    .expect_err("calling an EOA should fail");
+    let mut contract = Contract::preflight(Address::ZERO, &mut env);
+    contract
+        .call_builder(&ViewCallTest::testBlockhashCall {})
+        .call()
+        .expect_err("calling an EOA should fail");
 }
 
-fn eth_call<C>(view_call: ViewCall<C>, block: u64) -> C::Return
+/// Type just used in tests to avoid restriction of operating over builders with different generic
+/// `E` types.
+#[derive(Debug, Default)]
+struct BuilderOverrides {
```

Comment:
Hmm, I think it might be even more complicated than that:
We also need the ability to specify the `with_chain_spec'. I think this is actually a shortcoming in the current tests that we do not have.

Reply:
Yeah, that's definitely a negative of these new APIs, that testing host and guest in the same place is messy. Although for this, the chain spec is defined per test, so this isn't a regression, just makes things a bit more verbose/messy. Is this a dealbreaker?

Reply:
I'm afraid there is no other way if we want to follow alloy's `call_builder` pattern. Still, it was much nicer before.